core/services/wifi/WifiManager.py

```diff
@@ -70,20 +71,22 @@ def __dict_from_list(data: bytes) -> Dict[str, Any]:
 
         return output
 
-    def get_wifi_available(self) -> Union[List[Dict[str, Any]], str]:
+    def get_wifi_available(self) -> List[Dict[str, Any]]:
         """Get a dict from the wifi signals available"""
         self.wpa.send_command_scan()
         data, result = self.wpa.send_command_scan_results()
         if not result:
-            return "Failed to fetch wifi list"
+            warnings.warn(RuntimeWarning("Failed to fetch wifi list."))
+            return []
```

Comment:
Does it mean there's no way to tell if the wifi is failing or there are not networks in range?

Reply:
The warn is meant to do it.

The other option would be to raise an error an catch it later. Discussing with @patrickelectric we though it would be better to warn so the user should be able to continue using the program.

What do you think?

Reply:
It's a bit weird IMO using a warning log system as a return mechanism.. 

So for every call, we need to check for warnings.. if that's the plan here I don't see how is that different from a raise with a try block

Reply:
I was thinking about this, and yeah, there's no reason not to raise there. If the library is used by a higher-level app, it should try/catch the errors and answer accordingly.

Will move to raises.